From e464bab42cc0f08142513e6eee9f520f2a45266b Mon Sep 17 00:00:00 2001
From: Jeremy Rocher <jeremy.rocher@intel.com>
Date: Mon, 18 Apr 2016 17:27:00 +0200
Subject: [PATCH 0735/1214] intel_pstore_pram: load module earlier

Currently intel_pstore_pram is loaded as a normal module at
device_initcall() but if a panic occurs before it is loaded, nothing
is logged in pstore.

Load module earlier, in postcore_initcall(), like ramoops.

Signed-off-by: Jeremy Rocher <jeremy.rocher@intel.com>
---
 drivers/platform/x86/intel_pstore_pram.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/platform/x86/intel_pstore_pram.c b/drivers/platform/x86/intel_pstore_pram.c
index 2d5205b..78dd9c3b 100644
--- a/drivers/platform/x86/intel_pstore_pram.c
+++ b/drivers/platform/x86/intel_pstore_pram.c
@@ -125,7 +125,7 @@ static int __init intel_pram_init(void)
 
 	return register_pram_dev(pramt->addr, pramt->size);
 }
-device_initcall(intel_pram_init);
+postcore_initcall(intel_pram_init);
 
 static void __exit intel_pram_exit(void)
 {
-- 
2.7.4

